src/m365/spo/commands/listitem/listitem-batch-set.ts

```diff
@@ -0,0 +1,290 @@
+  }
+
+  private async sendBatchRequest(webUrl: string, requestBody: string): Promise<void> {
+    const requestOptions: CliRequestOptions = {
+      url: `${webUrl}/_vti_bin/client.svc/ProcessQuery`,
+      headers: {
+        'Content-Type': 'text/xml'
+      },
+      data: requestBody
+    };
+    const res: any = await request.post(requestOptions);
+    const json: ClientSvcResponse = JSON.parse(res);
+    const response: ClientSvcResponseContents = json[0];
+    if (response.ErrorInfo) {
+      throw response.ErrorInfo.ErrorMessage + " - " + response.ErrorInfo.ErrorValue;
+    }
+  }
+
+  private mapActions(index: number, row: any, fields: FieldDetails[], systemUpdate?: boolean): [string, number] {
+    const objectPathId = index;
+    let actionString = '';
+
+    fields.forEach((field: FieldDetails) => {
+      switch (field.TypeAsString) {
+        case 'UserMulti':
+          const userMultiString: string[] = row[field.InternalName].split(';').map((element: string) => {
+            return `<Object TypeId="{c956ab54-16bd-4c18-89d2-996f57282a6f}"><Property Name="Email" Type="Null" /><Property Name="LookupId" Type="Int32">${element}</Property><Property Name="LookupValue" Type="Null" /></Object>`;
+          });
+          actionString += `<Method Name="SetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter Type="Array">${userMultiString.join('')}</Parameter></Parameters></Method>`;
+          break;
+        case 'Lookup':
+          actionString += `<Method Name="SetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter TypeId="{f1d34cc0-9b50-4a78-be78-d5facfcccfb7}"><Property Name="LookupId" Type="Int32">${row[field.InternalName]}</Property><Property Name="LookupValue" Type="Null"/></Parameter></Parameters></Method>`;
+          break;
+        case 'LookupMulti':
+          const lookupMultiString: string[] = row[field.InternalName].split(';').map((element: string) => {
+            return `<Object TypeId="{f1d34cc0-9b50-4a78-be78-d5facfcccfb7}"><Property Name="LookupId" Type="Int32">${element}</Property><Property Name="LookupValue" Type="Null" /></Object>`;
+          });
+          actionString += `<Method Name="SetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter Type="Array">${lookupMultiString.join('')}</Parameter></Parameters></Method>`;
+          break;
+        default:
+          actionString += `<Method Name="ParseAndSetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter Type="String">${(<any>row)[field.InternalName].toString()}</Parameter></Parameters></Method>`;
```

Comment:
The code crashes on this line if you have an empty line in your CSV file.

Reply:
But why would we have an empty line in the CSV-file though 😄 No I get your comment, and will check 😉

Reply:
I have resolved this by changing the method `parseCsvToJson` in the `formatting` util, as I believe that we should never have an empty line at the end of our json array. An empty line would simple show as `''`, as you can see in the screenshot below. 
![image](https://user-images.githubusercontent.com/9193870/233789878-e3929a72-ce54-4192-a6ed-206d73faa089.png)
This could be solved by doing the following: 
![image](https://user-images.githubusercontent.com/9193870/233789900-a49738e5-a707-4fb5-9cea-724402d05f95.png)